### ISSUE TITLE
Allow again brew cask search?

### ISSUE THREAD

**Issue by kud** [Homebrew/homebrew-cask#53045]
### Description of feature/enhancement

Allow again `brew cask search`

### Justification

I was glad to see `cask` merged in the global `brew` command in order not to think so much if it's on the global repo or the cask repo when you search a package but I feel it's done in a half way.

Indeed, you now have to use `brew search` to find a package also on `cask`. But you can't use `brew (un)install xxx` for a cask package.

It seems not to be consistent. Which makes me think `brew cask seach` still should/could exist.

**Comment by reitermarkus** [comment]
If you don't know whether something is a cask, you would have to search first with `brew search`, not find it, and then use `brew cask search` to find it.

Allowing `brew cask search` again doesn't gain us anything.

**Comment by j0hnm4r5** [comment]
@reitermarkus What is the rationale for this change?

I've searched a bit, and I see issues surrounding the effect of the removal of `search`, but I don't see the issue where it must have been first discussed; could you point me in that direction? I'd love to fully understand why you guys removed the search command.

**Comment by j0hnm4r5** [comment]
I feel like what they're going for is to fully integrate `brew cask` into `brew`; deprecating `brew cask search` is a step toward that. The problem is that that integration isn't complete yet, so all of the commands haven't been migrated to the new format. Until that happens, users are left with an inconsistent and fragmented experience.

**Comment by reitermarkus** [comment]
> What is the rationale for this change?

The rationale is to remove duplicate code shared between Homebrew and Homebrew Cask, which in turn makes the project as a whole more maintainable.

**Comment by j0hnm4r5** [comment]
I totally get that, really, I do. But you can't disregard user experience to make that happen. Is there some way I can get involved with this sort of thing? I love Homebrew Cask, and I want to help.

**Comment by reitermarkus** [comment]
> But you can't disregard user experience to make that happen.

From my point of view this improves the experience by not needing two commands for the same thing.

> Is there some way I can get involved with this sort of thing?

You can look through the open issues and send PRs to improve or fix them.

**Comment by j0hnm4r5** [comment]
Speak to my previous comment then, about the inconsistencies in command usage:

**SEARCHING**: `brew search`

**INSTALLATION**: `brew cask install`

**INFO**: `brew cask info`

...

I agree that reducing the number of commands is a good thing, but you've only done it in one place (with search). Now users like myself and the OP have to remember that search _doesn't_ use cask, but everything else _does_. And presumably, as more of these commands are merged into standalone brew, we'll have to remember which are just brew and which are cask. And that burden shouldn't be on the consumer.

**TLDR**: I'm all for reducing the number of commands (both for maintainability and usability), but it's an all-or-nothing kind of thing. Inconsistency in which commands are `brew` and which are `brew cask` does more harm to the user than leaving them all alone.

**Comment by vitorgalvao** [comment]
From [my comment on the other issue](https://github.com/Homebrew/homebrew-cask/issues/53143#issuecomment-428264917):

> If we make brew cask search work again, we’ll need to add more code to separate brew search and brew cask search. We can’t simply alias the commands because it makes no sense that brew cask search would also search formulae but it’d also make no sense to remove casks from brew search at this point.

**Comment by reitermarkus** [comment]
> but you've only done it in one place

`brew cask cleanup` is gone too.

> And that burden shouldn't be on the consumer.

I think “burden” is a bit of a stretch when talking about remembering which command to use. We give users plenty of time to adapt by warning them that the command is deprecated.
